File: app/src/main/java/com/example/ikastyle/UI/LoadoutDeleteButton.java
package com.example.ikastyle.UI;

import android.content.Context;
import android.os.AsyncTask;
import android.util.AttributeSet;
import android.util.Pair;
import android.view.View;

import com.example.ikastyle.Common.Const.NumberPlace;
import com.example.ikastyle.Common.Util;
import com.example.ikastyle.Dao.GearSetDao;
import com.example.ikastyle.Dao.MainCategoryDao;
import com.example.ikastyle.Dao.WeaponMainDao;
import com.example.ikastyle.Database.AppDatabase;
import com.example.ikastyle.DatabaseView.WeaponMain;
import com.example.ikastyle.Entity.GearSet;
import com.example.ikastyle.Entity.MainCategory;
import com.example.ikastyle.R;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import org.w3c.dom.Attr;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.TreeMap;
import java.util.stream.Collectors;

public class LoadoutDeleteButton extends FloatingActionButton {
    private GearSet gearSet;

    public LoadoutDeleteButton(Context context, AttributeSet attrs){
        super(context, attrs);
    }

    public void setGearSet(GearSet gearSet) {
        this.gearSet = gearSet;
    }

    public GearSet getGearSet() {
        return gearSet;
    }
}

File: app/src/main/java/com/splatool/ikastyle/view/customizedView/LoadoutDeleteButton.kt
package com.splatool.ikastyle.view.customizedView

import android.content.Context
import android.util.AttributeSet
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.splatool.ikastyle.R
import com.splatool.ikastyle.model.data.entity.Loadout

class LoadoutDeleteButton(context: Context, attrs: AttributeSet?) :
    FloatingActionButton(context, attrs) {
    var loadout: Loadout? = null
    var loadoutId : Int = 0

    init{
        // カスタム属性gearKindの値を取得しフィールド変数gearKindに入れる
        val typedArray = getContext().obtainStyledAttributes(attrs, R.styleable.LoadoutDeleteButton, 0, 0)
        loadoutId = typedArray.getInt(R.styleable.LoadoutDeleteButton_loadoutId, 0)
    }
}